File: backend/entity/assistance.go
package entity

import (
	"gorm.io/gorm"
)

type Assistance struct {
	gorm.Model
	Massage string `json:"massage" valid:"required~massage is required"`
	
	ChatroomID uint `json:"chatroom_id" valid:"required~chatroom is required"`
	Chatroom Chatroom `gorm:"foreignKey:ChatroomID" json:"chatroom"`
	
	SenderID uint `json:"sender_id" valid:"required~sender is required"`
	Sender User	`gorm:"foreignKey:SenderID" json:"sender"`
	
}

File: frontend/src/interfaces/assistance.ts
import type { UserResponse } from "./user"
import type { ChatroomRespone } from "./chatroom"

export interface AssistanceRespone {
    ID: number
    massage: string
    CreatedAt: string

    sender_id: number
    sender: UserResponse

    chatroom_id: number
    chatroom: ChatroomRespone
}

export interface AssistanceCreate {
    massage: string

    sender_id: number
    chatroom_id: number

}

export interface AssistanceUpdate {
    ID: number
    massage?: string

    sender_id?: number
    chatroom_id?: number

}